test/es-module/test-vm-main-context-default-loader.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+const fixtures = require('../common/fixtures');
+const url = require('url');
+const fs = require('fs');
+const {
+  compileFunction,
+  Script,
+  createContext,
+  constants: { USE_MAIN_CONTEXT_DEFAULT_LOADER },
+} = require('vm');
+const assert = require('assert');
+
+common.expectWarning('ExperimentalWarning',
+                     'vm.USE_MAIN_CONTEXT_DEFAULT_LOADER is an experimental feature and might change at any time');
+
+assert(
+  !process.execArgv.includes('--experimental-vm-modules'),
+  'This test must be run without --experimental-vm-modules');
+
+assert.strictEqual(typeof USE_MAIN_CONTEXT_DEFAULT_LOADER, 'symbol');
+
+async function testNotFoundErrors(options) {
+  // Import user modules.
+  const script = new Script('import("./message.mjs")', options);
+  // Use try-catch for better async stack traces in the logs.
+  await assert.rejects(script.runInThisContext(), { code: 'ERR_MODULE_NOT_FOUND' });
+
+  const imported = compileFunction('return import("./message.mjs")', [], options)();
+  // Use try-catch for better async stack traces in the logs.
+  await assert.rejects(imported, { code: 'ERR_MODULE_NOT_FOUND' });
+}
+
+async function testLoader(options) {
+  {
+    // Import built-in modules
+    const script = new Script('import("fs")', options);
+    let result = await script.runInThisContext();
+    assert.strictEqual(result.constants.F_OK, fs.constants.F_OK);
+
+    const imported = compileFunction('return import("fs")', [], options)();
+    result = await imported;
+    assert.strictEqual(result.constants.F_OK, fs.constants.F_OK);
+  }
+
+  const modulePath = fixtures.path('es-modules', 'message.mjs');
+  fs.copyFileSync(modulePath, tmpdir.resolve('message.mjs'));
+
+  {
+    const namespace = await import(modulePath);
```

Comment:
I believe this is the last spot where the specifier should be a URL.

```suggestion
  const moduleUrl = fixtures.fileURL('es-modules', 'message.mjs');
  fs.copyFileSync(moduleUrl, tmpdir.resolve('message.mjs'));

  {
    const namespace = await import(moduleUrl);
```